### ISSUE TITLE
 .SF Pro Display and .SF Pro Text  renders same font on iOS.

### ISSUE THREAD

**Issue by SimonVillage** [flutter/flutter#82995]
```
Text('12,23123,123.123 asdASD',
    style: TextStyle(
      fontFamily: '.SF Pro Text',
    )),
Text('12,23123,123.123 asdASD',
    style: TextStyle(
      fontFamily: '.SF Pro Display',
    ))
```

Both widgets are displayed with the same Font on iOS. I also tried using `.SF UI Text` but that also does not make any difference.

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.0.6, on macOS 11.3.1 20E241 darwin-arm, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.56.2)
[✓] Connected device (2 available)

• No issues found!
```

Code I used
```flutter
import 'package:flutter/cupertino.dart';

void main() {
  runApp(CupertinoApp(
    title: 'Flutter Demo',
    home: MyHomePage(),
  ));
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
      child: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text('12,23123,123.123 asdASD',
                style: TextStyle(
                  fontFamily: '.SF Pro Text',
                )),
            Text('12,23123,123.123 asdASD',
                style: TextStyle(
                  fontFamily: '.SF Pro Display',
                )),
          ],
        ),
      ),
    );
  }
}

```

**Comment by darshankawar** [comment]
@SimonHausdorf 
Would like to confirm if you added the said font family in pubspec.yaml and still get the same behavior ?

Thanks.

**Comment by SimonHausdorf** [comment]
But isn't SF Pro the default font for Flutter on iOS?
https://github.com/flutter/flutter/blob/e7b7ebc066c1b2a5aa5c19f8961307427e0142a6/packages/flutter/lib/src/cupertino/text_theme.dart

Not sure why this font has to be added manually to use it?

**Comment by darshankawar** [comment]
That is correct. No need to add it manually and running your code sample I see the same behavior on latest stable (2.2.0)

<img width="240" alt="Screenshot 2021-05-20 at 7 32 56 PM" src="https://user-images.githubusercontent.com/67046386/118992534-5d888280-b9a2-11eb-90bd-d283f13f0fb1.png">

<details>
<summary> flutter doctor -v </summary>

```
[✓] Flutter (Channel stable, 2.2.0, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.2.0 at /Users/dhs/documents/fluttersdk/flutter
    • Framework revision b22742018b (5 days ago), 2021-05-14 19:12:57 -0700
    • Engine revision a9d88a4d18
    • Dart version 2.13.0

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 14.4.1
    • macOS (desktop)           • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 90.0.4430.212

! Doctor found issues in 1 category.


```

</details>

**Comment by SimonHausdorf** [comment]
The code also references `SF UI` which is the old style as I know. There is a related issue to that: https://github.com/flutter/flutter/issues/63507

**Comment by ch-muhammad-adil** [comment]
Is there any solution available here?

**Comment by sglim** [comment]
Workaround: Use letterSpacing: -2.0. With this option, it looks like `SF UI Display`.

**Comment by thomasmol** [comment]
I think i found the problem regarding this issue, check my comment on #63507.

**Comment by reinhart1010** [comment]
Hi, I'm still interested in this issue, already found a way which might solve this, but still don't understand the internal components Flutter or Dart is calling to render these fonts natively.

Perhaps best way to resolve this is by introducing font name aliases, similar to how it's commonly done with CSS (by aliasing `-apple-system`, `BlinkMacSystemFont`, `system-ui`, `ui-sans-serif`, etc.) Each target platform then associate the alias with their respective system fonts.

In the case of Apple platforms, each associated alias will link to [one of these four system constants (see WWDC19)](https://developer.apple.com/wwdc19/227?time=103) to render the appropriate font(s).

**Comment by MitchellGoodwin** [comment]
Fixed by [engine/pull/46857](https://github.com/flutter/engine/pull/46857).
